File: crud-item-list/src/test/java/com/gurunars/crud_item_list/CheckerMoveUpTest.java
package com.gurunars.crud_item_list;

import org.junit.Test;
import org.mockito.internal.util.collections.Sets;

import java.util.Arrays;
import java.util.List;

import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

public class CheckerMoveUpTest {

    private final ActionMoveUp<String> moveUpChecker = new ActionMoveUp<>();
    private final List<String> all = Arrays.asList("one", "two", "three", "four");

    @Test
    public void selectingFirstItem_leadsToFalse() throws Exception {
        assertFalse(moveUpChecker.canPerform(all, Sets.newSet("one")));
    }

    @Test
    public void selectingInterruptedChunk_leadsToFalse() throws Exception {
        assertFalse(moveUpChecker.canPerform(all, Sets.newSet("two", "four")));
    }

    @Test
    public void selectingSolidChunkBeforeLast_leadsToTrue() throws Exception {
        assertTrue(moveUpChecker.canPerform(all, Sets.newSet("two", "three", "four")));
    }

}

File: crud-item-list/src/test/kotlin/com/gurunars/crud_item_list/CheckerMoveUpTest.kt
package com.gurunars.crud_item_list

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import org.mockito.internal.util.collections.Sets
import java.util.*

class CheckerMoveUpTest {

    private val moveUpChecker = ActionMoveUp<StringItem>()
    private val all = listOf("one", "two", "three", "four")

    @Test
    @Throws(Exception::class)
    fun selectingFirstItem_leadsToFalse() {
        moveUpChecker.canPerform(all, setOf("one"), {assertFalse(it)})
    }

    @Test
    @Throws(Exception::class)
    fun selectingInterruptedChunk_leadsToFalse() {
        moveUpChecker.canPerform(all, setOf("two", "four"), {assertFalse(it)})
    }

    @Test
    @Throws(Exception::class)
    fun selectingSolidChunkBeforeLast_leadsToTrue() {
        moveUpChecker.canPerform(all, setOf("two", "three", "four"), {assertTrue(it)})
    }

}
